cloto/manager/AuthorizationManager.py

```diff
@@ -36,47 +39,96 @@ class AuthorizationManager():
     """
     myClient = None
     client = requests
+    session = None
 
-    def generate_adminToken(self, username, password, url):
-        """This method generates an admin token."""
+    def generate_session(self, username, password, tenant_id, tenant_name, user_domain_name, auth_api, url):
+        """
+        Init the variables related to authorization, needed to execute tests
+        :return: The auth token retrieved
+        """
+
+        cred_kwargs = {
+            'auth_url': url,
+            'username': username,
+            'password': password
+        }
+
+        # Currently, both v2 and v3 Identity API versions are supported
+        if auth_api == 'v2.0':
+            cred_kwargs['tenant_name'] = tenant_name
+        elif auth_api == 'v3':
+            cred_kwargs['user_domain_name'] = user_domain_name
+
+        # Instantiate a Password object
         try:
-            admin_client = self.myClient.Client(username=username, password=password, auth_url=url)
-            return admin_client.auth_token
-        except AuthorizationFailure as auf:
-            raise(auf)
-        except Unauthorized as unauth:
-            raise(unauth)
+            identity_package = 'keystoneclient.auth.identity.%s' % auth_api.replace('.0', '')
+            identity_module = __import__(identity_package, fromlist=['Password'])
+            password_class = getattr(identity_module, 'Password')
+            logger.debug("Authentication with %s", password_class)
+            credentials = password_class(**cred_kwargs)
+        except (ImportError, AttributeError) as e:
+                raise e
+        # Get auth token
+        logger.debug("Getting auth token for tenant %s...", tenant_id)
+        print cred_kwargs
+        try:
+            auth_sess = self.session.Session(auth=credentials, timeout=DEFAULT_REQUEST_TIMEOUT)
+            auth_token = auth_sess.get_token()
+            print("Admin token generated:" + auth_token)
 
-    def checkToken(self, admin_token, token, tenant_id, url):
+        except (KeystoneClientException, KeystoneConnectionRefused) as e:
+            logger.error("No auth token (%s)", e.message)
+            print e.message
+            raise(e)
+
+        return auth_token
+
+    def checkToken(self, admin_token, token, tenant_id, url, auth_api):
         """checks if a token is valid against a url using an admin token."""
-        print("Starting Authentication of token %s " % token)
+        logger.info("Starting Authentication of token %s " % token)
```

Comment:
Please let `logger` do the string interpolation, in case this logging level is enabled: `logger.info("Starting Authentication of token %s " , token)`